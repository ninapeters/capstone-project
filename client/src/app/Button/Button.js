import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { motion } from 'framer-motion'
import { clickAnimation } from '../../services/animationVariants'

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  isRedIcon: PropTypes.bool,
  isNoCircle: PropTypes.bool,
  isSmall: PropTypes.bool,
  children: PropTypes.element.isRequired,
}

export default function Button({
  onClick,
  disabled,
  isRedIcon,
  isNoCircle,
  isSmall,
  testId,
  children,
}) {
  return (
    <ButtonStyled
      variants={clickAnimation}
      initial="rest"
      whileTap="pressed"
      onClick={onClick}
      disabled={disabled}
      isRedIcon={isRedIcon}
      isNoCircle={isNoCircle}
      isSmall={isSmall}
      data-testid={`${testId}-button`}
    >
      {children}
    </ButtonStyled>
  )
}

const ButtonStyled = styled(motion.button)`
  background: ${(props) =>
    props.isNoCircle ? 'transparent' : 'var(--white-transparent-min)'};
  border: none;
  border-radius: ${(props) =>
    props.isNoCircle ? '34px 10px 10px 34px' : '50%'};
  box-shadow: ${(props) => (props.isNoCircle ? 'none' : 'var(--shadow-light)')};
  fill: ${(props) =>
    props.isRedIcon ? 'var(--cta-red)' : 'var(--spotify-green)'};
  height: ${(props) => (props.isSmall ? '50px' : '60px')};
  padding: 16px;
  width: ${(props) => (props.isSmall ? '50px' : '60px')};
  &:disabled {
    background: var(--spotify-green-transparent);
    fill: var(--white);
  }
  &:focus {
    outline: 0;
  }
`
