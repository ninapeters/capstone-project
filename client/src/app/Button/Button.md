Save button

```jsx
import { SaveIcon } from '../Icons/Icons'
;<Button onClick={() => {}}>
  <SaveIcon />
</Button>
```

Save button disabled

```jsx
import { SaveIcon } from '../Icons/Icons'
;<Button disabled onClick={() => {}}>
  <SaveIcon />
</Button>
```

Delete button

```jsx
import { CrossIcon } from '../Icons/Icons'
;<Button isRedIcon onClick={() => {}}>
  <CrossIcon />
</Button>
```
