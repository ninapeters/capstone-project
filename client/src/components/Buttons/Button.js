import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

const ButtonStyled = styled.button`
  background: var(--white-transparent-min);
  border: none;
  border-radius: 50%;
  box-shadow: var(--shadow-light);
  fill: ${(props) =>
    props.isDeleteButton
      ? 'var(--cta-red)'
      : 'var(--spotify-green-transparent)'};
  height: 60px;
  padding: 14px;
  width: 60px;
  &:disabled {
    fill: var(--spotify-green);
  }
  &:focus {
    outline: 0;
  }
`
const Button = (props) => <ButtonStyled {...props} />

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  isDeleteButton: PropTypes.bool,
  disabled: PropTypes.bool,
  children: PropTypes.element.isRequired,
}

export default Button
