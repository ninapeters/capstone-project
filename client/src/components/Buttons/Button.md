Save button

```jsx
import { SaveIcon } from '../Icons'
;<Button onClick={() => {}}>
  <SaveIcon />
</Button>
```

Save button disabled

```jsx
import { SaveIcon } from '../Icons'
;<Button disabled onClick={() => {}}>
  <SaveIcon />
</Button>
```

Delete button

```jsx
import { CrossIcon } from '../Icons'
;<Button isRedIcon onClick={() => {}}>
  <CrossIcon />
</Button>
```
