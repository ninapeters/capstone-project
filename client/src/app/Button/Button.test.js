import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import Button from './Button'
import { ReactComponent as DeleteIcon } from '../../assets/x.svg'

describe('Button', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Button onClick={() => {}}>
        <DeleteIcon />
      </Button>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('calls onClick with the correct id', () => {
    const onClickMock = jest.fn()
    const { getByRole } = render(
      <Button onClick={() => onClickMock('a')}>
        <DeleteIcon />
      </Button>
    )
    const button = getByRole('button')
    user.click(button)

    expect(onClickMock).toHaveBeenCalledWith('a')
  })
})
