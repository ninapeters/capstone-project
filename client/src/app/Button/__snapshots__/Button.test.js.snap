// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  background: var(--white-transparent-min);
  border: none;
  border-radius: 50%;
  box-shadow: var(--shadow-light);
  fill: var(--spotify-green);
  height: 60px;
  padding: 16px;
  width: 60px;
}

.c0:disabled {
  background: var(--spotify-green-transparent);
  fill: var(--white);
}

.c0:focus {
  outline: 0;
}

<button
  class="c0"
  data-testid="undefined-button"
  style="transform: none;"
>
  <svg>
    save.svg
  </svg>
</button>
`;
